src/main/kotlin/io/github/inductiveautomation/kindling/utils/TabStrip.kt

```diff
@@ -151,6 +177,32 @@ open class TabStrip : DnDTabbedPane() {
         )
     }
 
+    fun editTabTitle(tabIndex: Int) {
+        val textField = JTextField(
+            getTitleAt(tabIndex),
+        )
+
+        textField.addKeyListener(
```

Comment:
Medium: You need to add a focus listener - you can start editing one tab title, then click elsewhere in the application, and it'll stay a text field, even allowing multiple text fields to be active in tabs at once